modules/skparagraph/src/Run.cpp

```diff
@@ -62,7 +62,7 @@ void Run::calculateMetrics() {
     fCorrectAscent = fFontMetrics.fAscent - fFontMetrics.fLeading * 0.5;
     fCorrectDescent = fFontMetrics.fDescent + fFontMetrics.fLeading * 0.5;
     fCorrectLeading = 0;
-    if (SkScalarNearlyZero(fHeightMultiplier)) {
+    if (SkScalarNearlyEqual(fHeightMultiplier, 1.0f)) {
```

Comment:
Could you explain the logic of skipping the further calculations? Is it just for skipping calculations, and the previous logic was incorrect? Or is it to avoid division by zero?

Reply:
Previously `0` was a marker of "not overriden multiplier" to keep dscent/descent unchanged. Now, in case of `1` it's just optimization - this condition might be removed without any math effect. However I'd keep it to avoid added rounding error after these extra actions

Reply:
`fontSize == descent - ascent`, "leading" (gap between lines) is not a part of calculations here.
<img width="1596" height="899" alt="image" src="https://github.com/user-attachments/assets/69e281f6-5fdd-4f2e-9248-b753573895ae" />

So if `fHeightMultiplier  == 1` then `runHeight == fontIntrinsicHeight`.
It means that `fCorrectAscent`/`fCorrectDescent` won't be changed in both condition paths.

The note about `fBaselineShift` is valid - I'm not sure that it's correct to ignore it, but I don't want to change this logic here/now

Reply:
> If it is a bug, now it can be bugged a different case

Because of that, I lean towards removing this condition. It also looks like not necessary optimization, or even not optimization (checking condition can be more expensive than 10 math operations)

Reply:
If we wouldn't have this issue, I would lean towards keeping it as well, to reduce merge conflicts

Reply:
> checking condition can be more expensive than 10 math operations

Though we perform another condition check, so it is not our case

Reply:
Seems valid. I was probably wrong about `fLeading` - it's a part of `fFont.getSize()`.

Agree to remove this condition entirely then

Reply:
>Because of that, I lean towards removing this condition.

This would also mean that we need to understand how it affects on `fBaselineShift`, and if everything is alright after removing the condition. But I think we have to do this because of our logic change

Reply:
Text layout is badly covered, but probably I can write something based on getting position based on char's index. Will reply about that separately

Reply:
https://github.com/JetBrains/compose-multiplatform-core/pull/2254